include/zephyr/pm/device.h

```diff
@@ -581,6 +581,22 @@ int pm_device_power_domain_remove(const struct device *dev,
  * @retval false If device is not currently powered
  */
 bool pm_device_is_powered(const struct device *dev);
+
+/**
+ * @brief Setup a device driver into the lowest valid power mode
```

Comment:
I think we can remove "valid" here (unless there are invalid power modes, but this should be defined).
This being said it is unclear to me what would be the lowest power mode for a device. If s2ram is supported, could it be device in ram retention ?